spec/unit/shoulda/matchers/action_controller/permit_matcher_spec.rb

```diff
@@ -205,25 +293,31 @@
           for(:create)
       end
 
-      expect(&assertion).to fail_with_message(
-        'Expected controller to permit city and country, but it did not.'
-      )
+      message =
+        "Expected POST #create to restrict parameters to " +
+        ":name, :age, :city, and :country,\n" +
+        "but the restricted parameters were :name and :age instead."
+
+      expect(&assertion).to fail_with_message(message)
     end
   end
 
   describe 'negative failure message' do
     it 'includes all attributes that should not have been permitted but were' do
-      define_controller_for_resource_with_strong_parameters(action: :create) do
-        params.require(:user).permit(:name, :age)
+      define_controller_with_strong_parameters(action: :create) do
+        params.permit(:name, :age)
       end
 
       assertion = lambda do
         expect(controller).not_to permit(:name, :age).for(:create)
       end
 
-      expect(&assertion).to fail_with_message(
-        'Expected controller not to permit name and age, but it did.'
-      )
+      message =
+        "Expected POST #create not to restrict parameters to " +
+        ":name and :age,\n" +
+        "but it did."
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.